plugins/terrain_and_color/main.py

```diff
@@ -588,3 +591,83 @@ def get_identify_detail(self, point):
         self.refresh()
 
         return None
+
+    def get_zonal_stats_from_point(self, point: Vector3) -> List[Optional[Dict]]:
+        results = []
+        if self.boundary_data:
+            for i, data_role in enumerate(self.data_roles):
+                dtype, _ = data_role
+                if dtype is DataPlugin.RASTER:
```

Comment:
Since this is plugin-specific and we know the data roles, couldn't we just iterate over those roles specifically and never have to test this?

Reply:
For that matter couldn't you also iterate over the data instead of roles? For example:

```python
for plugin in (x for x in (self.terrain_data, self.attribute_data) if x is not None):
    ...
```

Reply:
Ah, yes we do that.

If we were having this logic outside of the plugin though, this would make sense. Make it so it doesn't have to be re-implemented for each plugin, but since this is the only plugin that does the work, then maybe it's okay for now.